test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -117,304 +120,380 @@ void doBlockchainTestNoLog(json_spirit::mValue& _v, bool _fillin)
-					continue;
-				}
-				catch (...)
-				{
-					cnote << testname + "state sync or block import did throw an exception\n";
-					checkJsonSectionForInvalidBlock(blObj);
-					continue;
-				}
+				importedBlocks.push_back(alterBlock);
+				importedBlocks.back().clearState(); //close the state as it wont be needed. too many open states would lead to exception.
+			}
+		}
+		catch (Exception const& _e)
+		{
+			cnote << testname + "block import throw an exception: " << diagnostic_information(_e);
+			checkExpectedException(blObj, _e);
+			eraseJsonSectionForInvalidBlock(blObj);
+		}
+		catch (std::exception const& _e)
+		{
+			cnote << testname + "block import throw an exception: " << _e.what();
+			cout << testname + "block import thrown std exeption" << std::endl;
+			eraseJsonSectionForInvalidBlock(blObj);
+		}
+		catch (...)
+		{
+			cout << testname + "block import thrown unknown exeption" << std::endl;
+			eraseJsonSectionForInvalidBlock(blObj);
+		}
 
-				//block from RLP successfully imported. now compare this rlp to test sections
-				BOOST_REQUIRE(blObj.count("blockHeader"));
+		blArray.push_back(blObj);  //json data
+		this_thread::sleep_for(chrono::seconds(1));
+	}//each blocks
 
-				//Check Provided Header against block in RLP
-				mObject emptyState;
-				TestBlock blockFromFields(blObj["blockHeader"].get_obj(), emptyState);
-				//ImportTransactions
-				BOOST_REQUIRE(blObj.count("transactions"));
-				for (auto const& txObj: blObj["transactions"].get_array())
-				{
-					TestTransaction transaction(txObj.get_obj());
-					blockFromFields.addTransaction(transaction);
-				}
+	if (_o.count("expect") > 0)
+	{
+		AccountMaskMap expectStateMap;
+		State stateExpect(State::Null);
+		ImportTest::importState(_o["expect"].get_obj(), stateExpect, expectStateMap);
+		if (ImportTest::compareStates(stateExpect, testChain.topBlock().state(), expectStateMap, Options::get().checkstate ? WhenError::Throw : WhenError::DontThrow))
+			if (Options::get().checkstate)
+				cerr << testname << endl;
+		_o.erase(_o.find("expect"));
+	}
 
-				// ImportUncles
-				if (blObj["uncleHeaders"].type() != json_spirit::null_type)
-					for (auto const& uBlHeaderObj: blObj["uncleHeaders"].get_array())
-					{
-						mObject uBlH = uBlHeaderObj.get_obj();
-						BOOST_REQUIRE((uBlH.size() == 16));
+	_o["blocks"] = blArray;
+	_o["postState"] = fillJsonWithState(testChain.topBlock().state());
+	_o["lastblockhash"] = "0x" + toString(testChain.topBlock().blockHeader().hash(WithSeal));
 
-						mObject emptyState;
-						TestBlock uncle(uBlH, emptyState);
-						blockFromFields.addUncle(uncle);
-					}
+	//make all values hex in pre section
+	State prestate(State::Null);
+	ImportTest::importState(_o["pre"].get_obj(), prestate);
+	_o["pre"] = fillJsonWithState(prestate);
 
-				checkBlocks(blockFromFields, blockFromRlp, testname);
+	for (auto iterator = chainMap.begin(); iterator != chainMap.end(); iterator++)
+		delete iterator->second;
```

Comment:
For safety I would reset the pointer like  `iterator->second = nullptr` after `delete`.  Somebody might find an already `delete`d pointer.